libvast/test/command.cpp

```diff
@@ -98,11 +127,21 @@ TEST(parsing both) {
 TEST(nested arg parsing) {
   auto cmd1 = root.add<foo>("foo");
   auto cmd2 = cmd1->add<bar>("bar");
-  exec("foo -v 42 bar -o 123");
+  exec("foo -v 42 bar -o 123 '--this should not -be parsed ! x'");
   CHECK_EQUAL(cmd1->value, 42);
   CHECK_EQUAL(cmd2->other_value, 123);
   CHECK_EQUAL(caf::deep_to_string(options),
               R"([("flag", false), ("other-value", 123), ("value", 42)])");
+  CHECK_EQUAL(cmd1->tested_proceed, true);
```

Comment:
I think this adds too much to the unit test `nested arg parsing` and instead deserves its own unit test.